charts/newrelic-infra-operator/values.yaml

```diff
@@ -0,0 +1,102 @@
+  # pullSecrets:
+    # - name: regsecret
+
+admissionWebhooksPatchJob:
+  image:
+    repository: jettech/kube-webhook-certgen
+    tag: v1.5.0
+    pullPolicy: IfNotPresent
+  # Volume mounts to add to the job, you might want to mount tmp if Pod Security Policies
+  # Enforce a read-only root.
+  volumeMounts: []
+  #  - name: tmp
+  #    mountPath: /tmp
+
+  # Volumes to add to the job container
+  volumes: []
+  #  - name: tmp
+  #    emptyDir: {}
+
+replicas: 1
+
+resources:
+  limits:
+    memory: 80M
+  requests:
+    cpu: 100m
+    memory: 30M
+
+serviceAccount:
+  # Specifies whether a ServiceAccount should be created for the job and
+  # the deployment.
+  create: true
+  # The name of the ServiceAccount to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name:
+
+# Configure podSecurityContext
+podSecurityContext:
+  enabled: false
+  fsGroup: 1001
+  runAsUser: 1001
+  runAsGroup: 1001
+
+# Use cert manager for webhook certs, rather than the built-in patch job
+certManager:
+  enabled: false
+
+# Use custom tls certificates for the webhook, or let the chart handle it
+# TODO: Add support for defining custom certs for the webhook here
+# customTLSCertificate: {}
+
+# If you wish to provide additional annotations to apply to the pod(s), specify
+# them here
+# podAnnotations:
+
+# Pod scheduling priority
+# Ref: https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/
+# priorityClassName: high-priority
+
+# Webhook timeout
+#
+# Configure how long the API server should wait for a webhook to respond before treating the call as a failure
+#
+# Ref: https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#timeouts
+timeoutSeconds: 30
+
+# Operator configuration
+# The following are the default values for the operator
+config:
+  # infraSidecarInjection: {}  # Sidecar injection can be disabled by setting it to an empty object
+
+  infraSidecarInjection:
+    policies:  # Map of policies to determine where to inject, indexed by namespace
+      "*":  # `*` defines a special default policy that will apply to all namespaces that do not have a specific entry
+        unlabeled: false  # Do not inject unless there is a label match
+        fargate: true  # Inject sidecar in fargate nodes
+        # List of additional labels that will be checked.
+        # Sidecar will be injected if label value is truthy, will not if it's falsy.
+        # Upon conflict (one label is truthy but other is falsy), the falsy takes preference.
+        # If none of the labels in this list exists, `unlabeled` will be honored.
+        labels:
+          - newrelic.com/inject-sidecar
+#      # Monitor everything in example-namespace unless it has `newrelic.com/inject: false`
+#      example-namespace:
+#        unlabeled: true
+#        labels:
+#          - newrelic.com/inject-sidecar
```

Comment:
To be honest, I'd go with more standard configuration, which would look like following:
```yaml
config:
  infraSidecarInjection:
    policies:
    # Inject to all pods in namespace "foo"
    - namespaceName: "foo"
    # Inject to all pods in namespace with label "foo=bar".
    - namespaceSelector:
        matchLabels:
          foo: bar
    # Inject to pods in all namespaces with label "foo=baz".
    - podSelector:
        matchLabels:
          foo: baz
```

Then `namespaceSelector` and `podSelector` fields will use [LabelSelector](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#LabelSelector), which allows to do any kind of selection (match, not match etc).

`namespaceName` is special for <1.21 clusters for matching the namespaces, which has no labels.

It will both simplify the implementation and will provide consistent API for users, similar to other Kubernetes resources.

Reply:
Agreed to go for the k8s standard, I didn't know this mechanism was as flexible. Does this miechanism support negative matches? I like the idea of being able to define a policy that achieves "monitor everything in this namespace EXCEPT this pods".

Not super critical, since you always can label everything else, but just wondering.

Reply:
You can use `notIn` operator for everything except as far as I understand: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements.